src/pandas_profiling/model/correlations.py

```diff
@@ -5,7 +5,12 @@
 import numpy as np
 import pandas as pd
 from multimethod import multimethod
-from pandas.core.base import DataError
+
+try:
```

Comment:
Instead of try/except, check using pd.__version__ (which is already available)
This prevents unnecessary work for older pandas versions.

Reply:
`pandas.__version__` is not being used for any other check in `pandas-profiling` so far. I'm not sure it would be wise to add it here.

With "unnecessary work", do you mean a single `except:` clause processing? That would be done exactly one over the runtime of a program, and takes a negligible time nevertheless...

Reply:
Not in the library - there was no need - but it is used in the tests, and in the type system (visions), e.g.:
```def pandas_version():
    return tuple(int(s) for s in pd.__version__.split("."))


@pytest.mark.skipif(
    pandas_version() <= (1, 1, 0), reason="requires pandas 1.1.1 or higher"
)
```

You're right that in this case performance would probably not matter much.  I prefer this syntax as it's more explict